tiled/adapters/sequence.py

```diff
@@ -0,0 +1,200 @@
+
+    def __init__(
+        self,
+        filepaths: List[Path],
+        *,
+        structure: Optional[ArrayStructure] = None,
+        metadata: Optional[JSON] = None,
+        specs: Optional[List[Spec]] = None,
+        access_policy: Optional[AccessPolicy] = None,
+    ) -> None:
+        """
+
+        Parameters
+        ----------
+        seq :
+        structure :
+        metadata :
+        specs :
+        access_policy :
+        """
+        self.filepaths = filepaths
+        # TODO Check shape, chunks against reality.
+        self.specs = specs or []
+        self._provided_metadata = metadata or {}
+        self.access_policy = access_policy
+        if structure is None:
+            dat0 = self._load_from_files(0)
+            shape = (len(self.filepaths), *dat0.shape[1:])
+            structure = ArrayStructure(
+                shape=shape,
+                # one chunk per underlying image file
+                chunks=((1,) * shape[0], *[(i,) for i in shape[1:]]),
+                # Assume all files have the same data type
+                data_type=BuiltinDtype.from_numpy_dtype(dat0.dtype),
+            )
+        self._structure = structure
+
+    @abstractmethod
+    def _load_from_files(
+        self, slc: Union[builtins.slice, int] = slice(None)
+    ) -> NDArray[Any]:
+        """Load the array data from files
+
+        Parameters
+        ----------
+        slc : slice
+            an optional slice along the left-most dimension in the resulting array; effectively selects a subset of
+            files to be loaded
+
+        Returns
+        -------
+            A numpy ND array with data from each file stacked along an addional (left-most) dimension.
+        """
+
+        pass
+
+    def metadata(self) -> JSON:
+        """
+
+        Returns
+        -------
+
+        """
+        # TODO How to deal with the many headers?
+        return self._provided_metadata
+
+    def read(self, slice: Optional[NDSlice] = ...) -> NDArray[Any]:
+        """Return a numpy array
+
+        Receives a sequence of values to select from a collection of image files
+        that were saved in a folder The input order is defined as: files -->
+        vertical slice --> horizontal slice --> color slice --> ... read() can
+        receive one value or one slice to select all the data from one file or
+        a sequence of files; or it can receive a tuple (int or slice) to select
+        a more specific sequence of pixels of a group of images.
+
+        Parameters
+        ----------
+        slice : NDSlice, optional
+            Specification of slicing to be applied to the data array
+
+        Returns
+        -------
+            Return a numpy array
+        """
+        if slice is Ellipsis:
+            arr = self._load_from_files()
+        elif isinstance(slice, int):
+            # e.g. read(slice=0) -- return an entire image (drop 0th dimension of the stack)
+            arr = self._load_from_files(slice)[-1]
```

Comment:
```suggestion
            arr = numpy.squeeze(self._load_from_files(slice), 0)
```

Using `squeeze` is slightly better because, if something unexpected happens, it will not silently drop data.